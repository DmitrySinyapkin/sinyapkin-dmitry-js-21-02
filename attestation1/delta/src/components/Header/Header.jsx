import React from "react";
import './Header.css';
import Logo from "../Logo/Logo";
import TopMenu from "../Menu/TopMenu";
import AuthorizationMenu from "../AuthorizationMenu/AuthorizationMenu";

const Header = () => {
    return (
        <div className="header">
            <Logo />
            <TopMenu />
            <AuthorizationMenu />
        </div>
    );
}

export default Header;
