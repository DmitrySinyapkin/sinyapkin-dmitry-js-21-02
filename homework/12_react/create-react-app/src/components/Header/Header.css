.header {
    width: 100%;
    padding: 20px 10px;
    background: #009a58;
}

.header__text {
    font-size: 30px;
    color: white;
}