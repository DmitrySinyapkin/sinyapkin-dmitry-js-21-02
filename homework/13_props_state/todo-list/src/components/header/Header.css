.header {
    width: 100%;
    height: 100%;
    padding: 10px;
    background: darkcyan;
}

.header h1 {
    color: white;
    font-size: 30px;
    text-align: center;
}
