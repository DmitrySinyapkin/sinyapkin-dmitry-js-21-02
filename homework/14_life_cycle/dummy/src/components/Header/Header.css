.header {
    padding: 20px 0px;
}

.header h1 {
    font-family: Arial;
    font-size: 55px;
    color: black;
}