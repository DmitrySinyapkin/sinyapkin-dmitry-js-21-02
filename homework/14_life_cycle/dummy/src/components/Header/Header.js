import React from "react";
import './Header.css';
import { ThemeContextConsumer } from "../../contexts/ThemeContext";

export class Header extends React.Component {
    render() {
        return (
            <ThemeContextConsumer>
                {
                    (context) => (
                        <div className="header">
                            <h1 className={`header__text ${context.darkTheme && "header__text_dark"}`}>{this.props.name}</h1>
                        </div>
                    )  
                }
            </ThemeContextConsumer>
        );
    }
}
