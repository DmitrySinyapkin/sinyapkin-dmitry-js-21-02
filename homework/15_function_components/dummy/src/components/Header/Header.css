.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.header__text {
    font-family: Arial;
    font-size: 55px;
    color: black;
}

.header__text_dark {
    color: white;
}
