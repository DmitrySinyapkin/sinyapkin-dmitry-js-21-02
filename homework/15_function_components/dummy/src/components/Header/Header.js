import React, { useContext } from "react";
import './Header.css';
import { ThemeContext } from "../../contexts/ThemeContext";
import { Menu } from 'antd';
import 'antd/dist/antd.css';

export const Header = (props) => {
    const themeContext = useContext(ThemeContext);
    return (
        <div className="header">
            <h1 className={`header__text ${themeContext.darkTheme && "header__text_dark"}`}>{props.name}</h1>
            <Menu mode="horizontal" onClick={props.onClick}>
                <Menu.Item key="users">Пользователи</Menu.Item>
                <Menu.Item key="reg">Регистрация</Menu.Item>
            </Menu>
        </div>
    );
};
