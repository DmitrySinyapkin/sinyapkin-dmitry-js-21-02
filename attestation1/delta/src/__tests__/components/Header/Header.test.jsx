import React from "react";
import { shallow } from 'enzyme';
import Header from '../../../components/Header/Header'

describe('Header', () => {
    test('snapshot', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render header children', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('div.header').children()).toHaveLength(3);
    });
});
