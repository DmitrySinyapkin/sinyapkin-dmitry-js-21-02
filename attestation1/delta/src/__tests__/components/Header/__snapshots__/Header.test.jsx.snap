// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className="header"
>
  <Logo />
  <TopMenu />
  <Connect(AuthorizationMenu) />
</div>
`;
